### PR TITLE
qt*-qttranslations: move qttools dependency to build

### PR DESCRIPTION
**Pull request by catap**
#### Description

qttranslations is a port which contains only .qm files; this ports is runtime dependency for near of all Qt-based applications like wireshark. qttools depends on clang because qdoc uses it to parce C++ files. That makes wireshark to depends on clang via qttranslations.

The first attempt to fix it at 43d7a354e89046be136d961c3bdb760788bb4224 simple moved definition of lib dependency from `qttools` to `qtX-qttools`, and here the real fix.

###### Type(s)

- [x] bugfix

###### Tested on

macOS 12.6.8 21G725 x86_64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by catap** [comment]
@mascguy can it be merged? I guess 5 days is clear maintainer timeout.

**Review by mascguy — APPROVED** [review]

**Comment by mascguy** [comment]
Kirill, since CI is skipped, I need to test some of these on a range of macOS releases.

And while you always do a very thorough job, I don't want to merge without some local validation.

In terms of an ETA, I'll try to get to it over the next few days.

**Comment by mascguy** [comment]
> Kirill, since CI is skipped, I need to test some of these on a range of macOS releases.
> 
> And while you always do a very thorough job, I don't want to merge without some local validation.
> 
> In terms of an ETA, I'll try to get to it over the next few days.

@chrstphrchvz can you take a look at these changes?

Even better, if you can quickly test building one or more of the translation subports, that would be awesome! Completely buried at the moment, and don't want to keep Kirill waiting for days.

**Comment by catap** [comment]
@mascguy I'm using qt5 via wireshark and the goal of that changes to remove useless clang from my everydays machine :)

**Comment by pmetzger** [comment]
Can we merge this now?

**Comment by mascguy** [comment]
> Can we merge this now?

Bear in mind that CI had to be skipped, as the sheer number of QT subports results in timeout.

I was hoping to test locally, but currently buried with other PRs.

So we can either wait for someone to test, or go ahead and merge as-is. Up to you...

**Comment by mascguy** [comment]
I'm presently on a lunch break from work, so I'll sanity-test a few of the `qt-translation` subports. More to follow shortly...

**Comment by chrstphrchvz** [comment]
> @chrstphrchvz can you take a look at these changes?
> 
> Even better, if you can quickly test building one or more of the translation subports, that would be awesome! Completely buried at the moment, and don't want to keep Kirill waiting for days.

I feel that all there is to “test” here is:

1. verify Kirill’s claim that these ports only install .qm files (which is shown at https://ports.macports.org/port/qt5-qttranslations/details/ etc. under “Port Health” > “(Files)” for each builder)
2. verify `port info qt5-qttranslations` etc. shows the correct revision and build/library dependencies.

Having confirmed these things, I agree with Kirill’s changes. Maybe I am just not paranoid enough, but in this case, I feel that completely rebuilding these ports myself will not show anything else useful. Maybe the more important thing missing from the PR checklist template, and which I have not checked myself, is whether dependents still work (particularly with qt5-qttools not active).

**Comment by mascguy** [comment]
Apart from the lib deps question - which is out-of-scope for this PR - the two I built look good. So let's merge.

Great work Kirill!

**Comment by catap** [comment]
Yahoo! Thanks, no more clang dependencies when installing Wireshark!

**Comment by chrstphrchvz** [comment]
One port which now needs qttools to build is qtkeychain: #21079

**Pull request merged**
